HW4_Zolotikov/protein_tool.py

```diff
@@ -0,0 +1,200 @@
+  """
+  Counts the number of amino acids with positive charge, negative charge, 
+  and neutral amino acids in the sequence.
+
+  Parameters
+  ----------
+  seq : str
+      Amino acid sequence (string).
+  positive_charge : list
+      List of amino acids with positive charge (default is ['R', 'K', 'H']).
+  negative_charge : list
+      List of amino acids with negative charge (default is ['D', 'E']).
+
+  Returns
+  -------
+  dict
+      A dictionary containing the counts of amino acids and their labels:
+      - 'Positive' for amino acids with positive charge.
+      - 'Negative' for amino acids with negative charge.
+      - 'Neutral' for neutral amino acids.
+  """
+  positive_count = 0
+  negative_count = 0
+  neutral_count = 0
+
+  for aa in seq:
+    if aa in positive_charge:
+      positive_count += 1
+    elif aa in negative_charge:
+      negative_count += 1
+    else:
+      neutral_count += 1
+    
+    result = {
+        'Positive': positive_count,
+        'Negative': negative_count,
+        'Neutral': neutral_count
+    }
+  return result
+
+
+# Ustin
+POLAR_AA = {'D', 'E', 'R', 'K', 'H', 'N', 'Q', 'S', 'T', 'Y', 'C'}
+NONPOLAR_AA = {'A', 'G', 'V', 'L', 'I', 'P', 'F', 'M', 'W'}
+DNA_AA = {'F': 'TTY', 'L': '(TTR or CTN)', 'I': 'ATH', 'M': 'ATG', 'V': 'GTN', 'S': '(TCN or AGY)', 'P': 'CCN', 'T': 'ACN', 'A': 'GCN',
+          'Y': 'TAY', 'H': 'CAY', 'Q': 'CAR', 'N': 'AAY', 'K': 'AAR', 'D': 'GAY', 'E': 'GAR', 'C': 'TGY', 'W': '(CGN or AGR)', 'R': 'AGY', 'G': 'GGN'}
+
+
+def define_polarity(seq: str) -> Dict[str, int]:
+    """
+    Counts polar and nonpolar aminoacids in aminoacid sequences.
+
+    Arguments:
+     str: sequence to count polar and nonpolar aminoacids.
+
+    Return:
+     Dict[str, int]: 
+          Dictionary with keys 'Polar', 'Nonpolar' and values of quantity of according groups in sequence. 
+    """
+    polarity_count = {'Polar': 0, 'Nonpolar': 0}
+    for aminoacid in seq:
+        if aminoacid in POLAR_AA:
+            polarity_count['Polar'] += 1
+        else:
+            polarity_count['Nonpolar'] += 1
+    return polarity_count
+
+
+def to_dna(seq: str) -> str:
+    """
+    Transforms aminoacid sequence to DNA sequence
+
+    Arguments
+    ---------
+     str: aminoacid sequence to transform to DNA sequence.
+
+    Return
+    ------
+     str: according DNA sequence.
+    """
+    sequence_dna = []
+    for aminoacid in seq:
+        sequence_dna.append(DNA_AA[aminoacid])
+    return ''.join(sequence_dna)
+
+
+#Margarita
+ABBREVIATION_THREE_TO_ONE = {'ALA':'A', 'CYS':'C', 'ASP':'D', 'GLU':'E', 'PHE':'F',
+                             'GLY':'G', 'HIS':'H', 'ILE':'I', 'LYS':'K', 'LEU':'L',
+                             'MET':'M', 'ASN':'N', 'PRO':'P', 'GLN':'Q', 'ARG':'R',
+                             'SER':'S', 'TRE':'T', 'VAL':'V', 'TRP':'W', 'TYR':'Y'}
+AMINO_ACIDS_ONE_LETTER = {'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K',
```

Comment:
А можно было бы объединить сеты :)
```suggestion
AMINO_ACIDS_ONE_LETTER = POLAR_AA.union(NONPOLAR_AA)
```